### DISCUSSION TITLE
Disable key binding when specific window is in focus

### DISCUSSION THREAD

**Discussion by C4K3** [i3/i3#5018] [Q&A]
I'd like to disable a specific key binding only when a specific window is in focus.

There are some discussions around this, for example https://faq.i3wm.org/question/7662/reverse-perl-matches-in-criteria-in-i3-config.1.html with the answer `bindsym $mod+Shift+q [con_id="__focused__" instance="^(?!VirtualBox).*$"] kill`, which as I understand works by having the command run against windows matching the given criteria, rather than when the focused window matches.

I'd like to do the same but with commands that don't run against a window, specifically switching workspace. The above config works, but if I change `kill` to `workspace 0` then it triggers regardless of what window is in focus.

Alternatively, is there a way to trigger a command when the focus changes? Like for_window but for focus changes, so I could have it change mode on a focus change.

**Comment by Airblader** [comment]
i3 doesn't have window-specific bindings. Even the workaround you found binds they key unconditionally, but then executes the command for each matched window (so if no window matches, never). 

You can listen for focus changes using the i3 IPC, for example, but there is nothing with pure configuration.
